luskan/src/main/java/org/opencadc/luskan/ws/StorageTapService.java

```diff
@@ -252,4 +254,17 @@ private String getState() {
         return ret;
     }
 
+    // Check that the TempStorageManager results directory is writable.
+    private void checkTempStorage() {
+        String tmpDir = LuskanConfig.getConfig().getFirstPropertyValue(LuskanConfig.TMPDIR_KEY);
+        Path tmpDirPath = Paths.get(tmpDir);
+        try {
+            Path tmpFile = Files.createTempFile(tmpDirPath, null,null);
+            Files.delete(tmpFile);
+        } catch (IOException e) {
+            throw new RuntimeException(String.format(
+                "Unable to write to TempStorageManager directory %s", tmpDirPath.toAbsolutePath()));
```

Comment:
In the filesystem storage adapter, the check for read-write permission fails with an IllegalArgumentException -- that is the general exception for bad input. RuntimeException should only be used for for a more extreme case like detected bug.